include/mc_control/MCController.h

```diff
@@ -48,6 +48,35 @@ struct MC_CONTROL_DLLAPI ControllerResetData
   const std::vector<std::vector<double>> q;
 };
 
+/** \brief Extra parameters that influence the creator construction */
+struct MC_CONTROL_DLLAPI ControllerParameters
+{
+  inline ControllerParameters() = default;
+  inline ControllerParameters(const ControllerParameters &) = default;
+  inline ControllerParameters(ControllerParameters &&) = default;
+  inline ControllerParameters & operator=(const ControllerParameters &) = default;
+  inline ControllerParameters & operator=(ControllerParameters &&) = default;
+
+#define ADD_PARAMETER(TYPE, NAME, DEFAULT) \
+  TYPE NAME##_ = DEFAULT;                  \
+  ControllerParameters & NAME(TYPE value)  \
+  {                                        \
+    NAME##_ = value;                       \
+    return *this;                          \
+  }
+  /** Backend used by this controller */
+  ADD_PARAMETER(mc_solver::QPSolver::Backend, backend, mc_solver::QPSolver::Backend::Tasks)
+  /** Whether to automatically load the robots' specific configuration (true by default) */
+  ADD_PARAMETER(bool, load_robot_config, true)
+  /** Where to load the robots' configuration config("robots") by default */
+  ADD_PARAMETER(std::vector<std::string>, load_robot_config_into, {"robots"})
+  /** Use the module name to find robot's specific values if true (default), use the robot's name otherwise */
+  ADD_PARAMETER(bool, load_robot_config_with_module_name, true)
```

Comment:
I believe this breaks the default behaviour of having a `robots` entry into the FSM configuration:

```
robots:
  panda1:
	module: panda_hand
        init_pos:
           translation: [0.2, 0, 0]
  panda2:
	module: panda_hand
        init_pos:
           translation: [0.8, 0, 0]
```

Will no longer load the init pos for either robots. Renaming the robot to `panda_hand` does move the robot init_pos, but we cannot do that for multiple robots:

```
robots:
  panda_hand: #this init pos gets loaded
	module: panda_hand
        init_pos:
           translation: [0.2, 0, 0]
  panda2: # but this one does not
	module: panda_hand
        init_pos:
           translation: [0.8, 0, 0]
```

Reply:
Hi @arntanguy 

Do you have an example where it actually breaks? I tried locally and it works correctly

Note that this PR only allows to load extra files into specific location of the configuration and does not change how the configuration is handled by any controller (including base controller).